tests/cuda/test_cuda_containers.cpp

```diff
@@ -0,0 +1,61 @@
+/** VecMem project, part of the ACTS project (R&D line)
```

Comment:
Splitting this executable into 2 source files seems to just be overcomplicating things. To do this properly you would also need to introduce a 3rd file, `test_cuda_containers_kernels.hpp`. But why do the split in the first place? What was the problem with having all the test code compiled by `nvcc`? :confused:

Reply:
This is unfortunately related to the C++14 requirement of the CUDA code. As we discussed earlier, `nvcc` won't compile many of the C++17 features we use here, so we need to compartmentalize. It seemed easier to put the CUDA code into its own little pocket like this than to fence off all the rest of the code, just in terms of implementation effort. That would also impact the code for people not using the CUDA plugin at all! I think it's a good idea to minimize the use of `nvcc` as much as possible, don't you agree?

Reply:
I see.

In that case we need a header file as well. :stuck_out_tongue: